### ISSUE TITLE
ezrecon: result "unknown"?

### ISSUE THREAD

**Issue by cloatre** [AFITWiSec/EZ-Wave#10]
Hello,

I am trying python tools to test my box VeraPlus and a wallplug Fibaro.

The network is detected:
=========================================================================
root@kali:~/EZ-Wave/tools# python ezstumbler.py 
[+] Found new Zwave network: 0xeb41dcac
[+][+] Found new Zwave node: 0xeb41dcac node 1
[+][+] Found new Zwave node: 0xeb41dcac node 2
==========================================================================

Then, when I try ezrecon to get more information, I always get this:
==========================================================================
root@kali:~/EZ-Wave/tools# python ezrecon.py 0xeb41dcac 2
Interrogating 0xeb41dcac Node 2

****************** Recon Results *********************

**************** Home ID: 0xeb41dcac *****************
NodeID 2:
	Manufacturer: Unknown 
	Product Name: Unknown 
	Library Type: Unknown
	Protocol Version/Subverion: Unknown
	Application Version/Subversion: Unknown
	Supported Command Classes: Unknown
	Basic Status: Unknown
	Configurations: 
		Skipped (use: ezrecon.py --homeid=0xeb41dcac --nodeid=2 --configs)
==========================================================================

Is-it normal?

==========================================================================
root@kali:~/EZ-Wave/tools# python ezfingerprint.py  0xeb41dcac 2
Scanning 2 with standard preamble
Active Scan of 0xeb41dcac for 10 seconds...
Node 2 found
Scanning 2 with shortened preamble
Active Scan of 0xeb41dcac for 10 seconds...
root@kali:~/EZ-Wave/tools# 0xeb41dcac Node 2: SD3501 Transceiver
==========================================================================

Thanks,

**Comment by gitjhall** [comment]
@cloatre You mentioned in another post that you are using only 1 HackRF. If that's still the case, you will only be able to transmit OR receive, not both. 

`ezstumbler` supports 2 types of scans: passive and active. The passive scan is simply listening for a z-wave device to talk. The active scan involved transmitting the z-wave equivalent of a broadcast ping to a target network. If you're only using 1 HackRF, you can only do the passive scan and you will only be able to identify devices that happen to talk while you're sniffing.

`ezrecon` and `ezfingerprint` both rely on transmitting packets to a target device and then capturing/parsing the response. When you run either without a 2nd HackRF to do the transmitting, your HackRF is just sitting there waiting for a response that will never come. That's why you see "Unknown" when you run `ezrecon`.

Basically, if you only have 1 HackRF you'll be limited to only passive z-wave sniffing.

**Comment by cloatre** [comment]
@gitjhall I have got a second hackRf today :-)
So i try again the python script and still get same results, I will check wireshark logs.

--------------------
root@kali:~/EZ-Wave/tools# hackrf_info 
Found HackRF board 0:
USB descriptor string: 000000000000000014d463dc2f4375e1
Board ID Number: 2 (HackRF One)
Firmware Version: 2015.07.2
Part ID Number: 0xa000cb3c 0x0056475f
Serial Number: 0x00000000 0x00000000 0x14d463dc 0x2f4375e1

Found HackRF board 1:
Board ID Number: 2 (HackRF One)
Firmware Version: 2014.08.1
Part ID Number: 0xa000cb3c 0x007e4755
Serial Number: 0x00000000 0x00000000 0x14d463dc 0x2f3b23e1

-------------------

root@kali:~/EZ-Wave/tools# python ezstumbler.py
WARNING: No route found for IPv6 destination :: (no default route?)
Sniffing for 60 seconds...
[+] Found new Zwave network: 0xeb41dcac
[+][+] Found new Zwave node: 0xeb41dcac node 1
[+][+] Found new Zwave node: 0xeb41dcac node 2
Active Scan of 0xea41dcac for 60 seconds...

****************** Scan Results *********************

*************** Home ID: 0xeb41dcac *****************
	NodeID 1
	NodeID 2

*****************************************************

*****************************************************



------------------

root@kali:~/EZ-Wave/tools# python ezrecon.py 0xeb41dcac 2
WARNING: No route found for IPv6 destination :: (no default route?)
Interrogating 0xea41dcac Node 2

****************** Recon Results *********************

**************** Home ID: 0xeb41dcac *****************
NodeID 2:
	Manufacturer: Unknown 
	Product Name: Unknown 
	Library Type: Unknown
	Protocol Version/Subverion: Unknown
	Application Version/Subversion: Unknown
	Supported Command Classes: Unknown
	Basic Status: Unknown
	Configurations: 
		Skipped (use: ezrecon.py --homeid=0xeb41dcac --nodeid=2 --configs)

-------------------

root@kali:~/EZ-Wave/tools# python ezfingerprint.py  0xeb41dcac 1
WARNING: No route found for IPv6 destination :: (no default route?)
Scanning 1 with standard preamble
Active Scan of 0xeb41dcac for 10 seconds...
Node 1 found
Scanning 1 with shortened preamble
Active Scan of 0xeb41dcac for 10 seconds...
root@kali:~/EZ-Wave/tools# 0xeb41dcac Node 1: SD3501 Transceiver
Exit

---------------------

root@kali:~/EZ-Wave/tools# python ezfingerprint.py  0xeb41dcac 2
WARNING: No route found for IPv6 destination :: (no default route?)
Scanning 2 with standard preamble
Active Scan of 0xeb41dcac for 10 seconds...
Node 2 found
Scanning 2 with shortened preamble
Active Scan of 0xeb41dcac for 10 seconds...
root@kali:~/EZ-Wave/tools# 0xeb41dcac Node 2: SD3501 Transceiver
Exi

----------------------------------

**Comment by gitjhall** [comment]
@cloatre First thing, you need to update the firmware on your 2nd HackRF

You've shown that the OS sees both devices but you need to make sure your flowgraph sees both. Make sure you re-enabled the transmit portion of the flowgraph (you mentioned before you had disabled it). Then make sure the packet sink block is referencing the 2nd HackRF. I always had issues using the device index and always used the serial number instead. Finally, do tests running the flowgraph directly in GRC and see if you get any errors loading either HackRF.

**Comment by cloatre** [comment]
@gitjhall 

I have updated my second HackRf this morning and I change the top_block file to put my hackRf Id and I have still result "unknown"
maybe there is a config file to give?

-------------------------------------
root@kali:~/EZ-Wave/tools# hackrf_info 
Found HackRF board 0:
USB descriptor string: 000000000000000014d463dc2f3b23e1
Board ID Number: 2 (HackRF One)
Firmware Version: 2015.07.2
Part ID Number: 0xa000cb3c 0x007e4755
Serial Number: 0x00000000 0x00000000 0x14d463dc 0x2f3b23e1

Found HackRF board 1:
USB descriptor string: 000000000000000014d463dc2f4375e1
Board ID Number: 2 (HackRF One)
Firmware Version: 2015.07.2
Part ID Number: 0xa000cb3c 0x0056475f
Serial Number: 0x00000000 0x00000000 0x14d463dc 0x2f4375e1
-----------------------------

**Comment by gitjhall** [comment]
Not sure what top_block file you're talking about. You should be modifying the Device Arguments field of the osmocom Sink block in grc.
